src/services/item/plugins/file/service.ts

```diff
@@ -92,80 +88,101 @@ class FileItemService {
       await validatePermission(repositories, PermissionLevel.Write, actor, item);
     }
 
-    const promises = files.map(async (fileObject) => {
-      const { filename, mimetype, fields, filepath: tmpPath, description } = fileObject;
-      const file = fs.createReadStream(tmpPath);
-      const { size } = fs.statSync(tmpPath);
+    // upload file one by one
+    // TODO: CHUNK FOR PERFORMANCE
+    const items: Item[] = [];
+    for await (const fileObject of files) {
+      const { filename, mimetype, fields, file: stream } = fileObject;
       const filepath = this.buildFilePath(); // parentId, filename
+      const i = await this._upload(actor, repositories, {
+        filepath,
+        parentId,
+        filename,
+        mimetype,
+        fields,
+        stream,
+      });
+      items.push(i);
+    }
 
-      // compute body data from file's fields
-      if (fields) {
-        const fileBody = Object.fromEntries(
-          Object.keys(fields).map((key) => [
-            key,
-            (fields[key] as unknown as { value: string })?.value,
-          ]),
-        );
-      }
-      // check member storage limit
-      await this.checkRemainingStorage(actor, repositories, size);
-
-      return await this.fileService
-        .upload(actor, {
-          file,
-          filepath,
-          mimetype,
-          size,
-        })
-        .then(() => {
-          return { filepath, filename, size, mimetype, uploaded: fileObject, description };
-        })
-        .catch((e) => {
-          throw e;
-        });
+    return items;
+  }
+
+  async _upload(
+    actor,
+    repositories,
+    {
+      description,
+      filepath,
+      parentId,
+      filename,
+      mimetype,
+      fields,
+      stream,
+    }: {
+      description?: string;
+      filepath;
+      parentId?: string;
+      filename;
+      mimetype;
+      fields?: MultipartFields;
+      stream: ReadableStream;
+    },
+  ) {
+    // compute body data from file's fields
+    if (fields) {
+      const fileBody = Object.fromEntries(
+        Object.keys(fields).map((key) => [
+          key,
+          (fields[key] as unknown as { value: string })?.value,
+        ]),
+      );
+    }
```

Comment:
Not sure why this is used for? Can `fileBody` be read from outside the if scope?